SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialEventTracer.cpp

```diff
@@ -256,16 +269,36 @@ void SpatialEventTracer::UpdateComponent(const Worker_ComponentUpdateOp& Op, con
 	StoredSpanIds.Push(SpanId);
 }
 
-TArray<FSpatialGDKSpanId> SpatialEventTracer::GetSpansForComponent(const EntityComponentId& Id) const
+void SpatialEventTracer::CommandRequest(const Worker_CommandRequestOp& Op, const FSpatialGDKSpanId& SpanId)
+{
+	FSpatialGDKSpanId& StoredSpanId = RequestSpanIds.FindOrAdd({ Op.request_id });
+	checkf(StoredSpanId.IsNull(), TEXT("CommandResponse received multiple times for request id %lld"), Op.request_id);
+	StoredSpanId = SpanId;
+}
+
+TArray<FSpatialGDKSpanId> SpatialEventTracer::GetAndConsumeSpansForComponent(const EntityComponentId& Id)
 {
 	const TArray<FSpatialGDKSpanId>* StoredSpanIds = EntityComponentSpanIds.Find(Id);
 	if (StoredSpanIds == nullptr)
 	{
 		return {};
 	}
+	EntityComponentsConsumed.Push(Id); // Consume on frame boundary instead, as these can have multiple uses.
 	return *StoredSpanIds;
 }
 
+FSpatialGDKSpanId SpatialEventTracer::GetAndConsumeSpanForRequestId(Worker_RequestId RequestId)
+{
+	const FSpatialGDKSpanId* SpanId = RequestSpanIds.Find(RequestId);
```

Comment:
Just interested to know if this ever actually starts to increase in size. I would hope we consume each request id we receive but might be good to check?

Reply:
Will check!

Reply:
Was all good - didn't seem to queue up when tracing cross-server!